lib/number-to-words.hoon

```diff
@@ -0,0 +1,105 @@
+  ?:  =(num 13)  [~ "thirteen"]
+  ?:  =(num 14)  [~ "fourteen"]
+  ?:  =(num 15)  [~ "fifteen"]
+  ?:  =(num 16)  [~ "sixteen"]
+  ?:  =(num 17)  [~ "seventeen"]
+  ?:  =(num 18)  [~ "eighteen"]
+  ?:  =(num 19)  [~ "nineteen"]
+  ::  20-99
+  =/  tpl  (div num ten)                                ::  tens place
+  =/  rem  (mod num ten)                                ::  ones place
+  =/  sfx                                               ::  suffix
+    ?:  |(=(rem 0) (gte tpl 10))
+      ~
+    ['-' (need $(num rem))]
+  ?:  =(tpl 2)  [~ (weld "twenty" sfx)]
+  ?:  =(tpl 3)  [~ (weld "thirty" sfx)]
+  ?:  =(tpl 4)  [~ (weld "fourty" sfx)]
+  ?:  =(tpl 5)  [~ (weld "fifty" sfx)]
+  ?:  =(tpl 6)  [~ (weld "sixty" sfx)]
+  ?:  =(tpl 7)  [~ (weld "seventy" sfx)]
+  ?:  =(tpl 8)  [~ (weld "eighty" sfx)]
+  ?:  =(tpl 9)  [~ (weld "ninety" sfx)]
+  ::  100-max
+  =/  num-break                                         ::  repeated pattern
+    ::  min: minimum to qualify for this break
+    ::  str: english word for this break
+    ::  sep: separator between this and the previous break
+    |=  [min=@u str=tape sep=tape]
+    =/  rem  (mod num min)
+    :-  ~
+    %+  weld
+      (weld (need ^$(num (div num min))) [' ' str])
+    ?:(=(rem 0) ~ (weld sep (need ^$(num rem))))
+  ?:  (lth num one-thousand)  (num-break one-hundred "hundred" " and ")
+  ?:  (lth num one-million)  (num-break one-thousand "thousand" ", ")
+  ?:  (lth num one-billion)  (num-break one-million "million" ", ")
+  ?:  (lth num one-trillion)  (num-break one-billion "billion" ", ")
+  ?:  (lth num one-quadrillion)  (num-break one-trillion "trillion" ", ")
+  ?:  (lth num one-quintillion)  (num-break one-quadrillion "quadrillion" ", ")
+  ?:  (lth num one-sextillion)  (num-break one-quintillion "quintillion" ", ")
+  ?:  (lth num one-septillion)  (num-break one-sextillion "sextillion" ", ")
+  ?:  (lth num one-octillion)  (num-break one-septillion "septillion" ", ")
+  ?:  (lth num one-nonillion)  (num-break one-octillion "octillion" ", ")
+  ?:  (lth num one-decillion)  (num-break one-nonillion "nonillion" ", ")
+  ?:  (lth num one-undecillion)  (num-break one-decillion "decillion" ", ")
+  ?:  (lth num one-duodecillion)  (num-break one-undecillion "undecillion" ", ")
+  ?:  (lth num one-tredecillion)  (num-break one-duodecillion "duodecillion" ", ")
+  ?:  (lth num one-quattuordecillion)  (num-break one-tredecillion "tredecillion" ", ")
+  ?:  (lth num one-quindecillion)  (num-break one-quattuordecillion "quattuordecillion" ", ")
+  ?:  (lth num one-sexdecillion)  (num-break one-quindecillion "quindecillion" ", ")
+  ?:  (lth num one-septendecillion)  (num-break one-sexdecillion "sexdecillion" ", ")
+  ?:  (lth num one-octodecillion)  (num-break one-septendecillion "septendecillion" ", ")
+  ?:  (lth num one-novemdecillion)  (num-break one-octodecillion "octodecillion" ", ")
+  ?:  (lth num one-vigintillion)  (num-break one-novemdecillion "novemdecillion" ", ")
```

Comment:
Some of these lines near the end are >80 characters. We try to stay <80 throughout the codebase.